File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPInvulnerable.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.entity.Player;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPInvulnerable implements PlayerStateProperty<Boolean> {
    @Override
    public String getConfigName() {
        return "invulnerable";
    }

    @Override
    public Boolean getDefaultSerializedProperty() {
        return false;
    }

    @Override
    public Boolean getSerializedPropertyFromPlayer(Player player) {
        return player.isInvulnerable();
    }

    @Override
    public void restoreProperty(Player player, Boolean invulnerable) {
        player.setInvulnerable(invulnerable);
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPInvulnerable.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.entity.Player
import org.noxet.noxetserver.playerstate.PlayerStateProperty

object PSPInvulnerable extends PlayerStateProperty[Boolean]:
  override def getConfigName: String = "invulnerable"

  override def getDefaultSerializedProperty: Boolean = false

  override def getSerializedPropertyFromPlayer(player: Player): Boolean =
    player.isInvulnerable

  override def restoreProperty(player: Player, invulnerable: Boolean): Unit =
    player.setInvulnerable(invulnerable)